src/core/ucc_coll.c

```diff
@@ -58,6 +58,19 @@ static ucc_status_t ucc_check_coll_args(const ucc_coll_args_t *coll_args,
                                            coll_args->dst.info);
         }
         break;
+    case UCC_COLL_TYPE_BCAST:
+        if (UCC_IS_INPLACE(*coll_args)) {
+            ucc_warn("Inplace not supported in bcast");
+        }
+        break;
+    case UCC_COLL_TYPE_BARRIER:
```

Comment:
why barrier doesn't support persistent? I guess same is true for fanin and fanout

Reply:
barrier (and fanin, fanout) supports persistent inside UCC collectives, but not in mpi test since it is not implemented in the test as a single collective test. 
And in general we dont have fanin/fanout mpi tests.

Reply:
makes sense, but then in core we should check only for inplace flag since persistent is ok to use

Reply:
removed persistent check in core